scriptworker_client/src/scriptworker_client/github_client.py

```
@@ -210,19 +210,13 @@ async def get_file_listing(self, path: str = "", branch: Optional[str] = None, d
 
         branch = branch or "HEAD"
 
-        query = Template(
-            dedent(
-                """
-            query RepoFiles {
-              repository(owner: "$owner", name: "$repo") {
-                object(expression: "$branch:$path") {
-                  $file_expr
-                }
-              }
-            }
-                """
-            )
-        )
+        if isinstance(paths, list):
```

Was this block added to handle nested calls where there may not be any subentries? If so, can we avoid making the nested call in that case instead?

It would be preferable to let an exception bubble up for any outside caller that doesn't provide any paths to fetch listings for; calling without a path to fetch for is not a valid invocation.

---

raised expection for the outside called